crates/core/component/compact-block/src/component/manager.rs

```diff
@@ -67,12 +67,16 @@ trait Inner: StateWrite {
             (None, Vec::new())
         };
 
-        let fmd_parameters = if height == 0 {
-            Some(
-                self.get_current_fmd_parameters()
-                    .await
-                    .context("could not get FMD parameters")?,
-            )
+        let previous_fmd_params = self
+            .get_previous_fmd_parameters()
+            .await
+            .context("could not get FMD parameters")?;
+        let current_fmd_params = self
+            .get_current_fmd_parameters()
+            .await
+            .context("could not get FMD parameters")?;
+        let fmd_parameters = if previous_fmd_params != current_fmd_params {
```

Comment:
```suggestion
        let fmd_parameters = if previous_fmd_params != current_fmd_params && current_fmd_params.as_of_block_height == height {
```

Reply:
e.g. if the params are:
```
0 0 0 0 0 0 1 1 1 1 1 1
```
we only want to send over the new params in the compact block where they changed